src/components/sidebar/MarkerSidebar.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+
+import explicitConnect from '../../utils/connect';
+import { selectedThreadSelectors } from '../../reducers/profile-view';
+import { getSelectedThreadIndex } from '../../reducers/url-state';
+import { formatMilliseconds } from '../../utils/format-numbers';
+import { CanSelectContent } from './index';
+
+import type {
+  ConnectedProps,
+  ExplicitConnectOptions,
+} from '../../utils/connect';
+import type { ThreadIndex } from '../../types/profile';
+import type {
+  IndexIntoTracingMarkers,
+  TracingMarker,
+} from '../../types/profile-derived';
+
+type SidebarDetailProps = {|
+  +label: string,
+  +value: React.Node,
+|};
+
+function SidebarDetail({ label, value }: SidebarDetailProps) {
+  return (
+    <React.Fragment>
+      <div className="sidebar-label">{label}:</div>
+      <div className="sidebar-value">{value}</div>
+    </React.Fragment>
+  );
+}
+
+type StateProps = {|
+  +selectedNodeIndex: IndexIntoTracingMarkers,
+  +selectedThreadIndex: ThreadIndex,
+  +markers: TracingMarker[],
```

Comment:
I think that instead of `selectedNodeIndex` and `markers` you could take `marker: TracingMarker | null` directly, that would simplify a bit the code below.

Retrieving the`marker` from the marker index and the `markers` array would be done in  `mapStateToProps`.

I see that the thread index is necessary for `MarkerTooltipContents` though so you can keep it.